libnavigation-copilot/src/main/java/com/mapbox/navigation/copilot/MapboxCopilotImpl.kt

```diff
@@ -303,7 +299,19 @@ internal class MapboxCopilotImpl(
         if (hasFeedback || !shouldSendHistoryOnlyWithFeedback) {
             endedAt = currentUtcTime()
             val diffTime = SystemClock.elapsedRealtime() - startSessionTime
-            push(DriveEndsEvent(DriveEnds(DriveEndsType.CanceledManually.type, diffTime)))
+            val driveEndsType = when {
+                arrivedAtFinalDestination -> when (deviceType) {
+                    DeviceType.HANDHELD -> DriveEndsType.Arrived
+                    DeviceType.AUTOMOBILE -> DriveEndsType.VehicleParked
```

Comment:
Interesting. Why are we differentiating by device type? Is that per spec?

Reply:
In spec we have the following:
- **vehicle_parked**, for in car apps when car was parked near the destination point
canceled_manually, for cases when user manually cancel navigation
- **arrived**, for mobile apps when user arrives the destination

To me it looks like in both cases we arrive at final destination, the only difference is whether it is a car app or a mobile app. Also the same logic is used in our app that we are migrating to Copilot.

Reply:
This is the closest way I found to match the spec without asking developers to provide event type manually.

Reply:
Sounds good to me. Thanks for clarifying 👍